File: LeetCode Practice Problems/728- Self Dividing Numbers.cpp
class Solution {
public:
    bool check(int n)
    {
        string s = to_string(n);
        for(char c : s)
            if(c == '0' || n%(c-'0') != 0)
                return false;
        
        return true;
    }
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> ans;
        for(int i=left;i<=right;i++)
        {
            if(check(i))
                ans.push_back(i);
        }
        
        return ans;
    }
};

File: LeetCode/LeetCode Practice Problems/728-Self Dividing Numbers.java
class Solution {
    
    public boolean find(int n) {
        for(char c: String.valueOf(n).toCharArray()) {
            if(c == '0' || (n%(c-'0') > 0)) {
                return false;
            }
        }
        return true;
    }
    
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> ans = new ArrayList();
        for(int i=left;i<=right;i++) {
            if(find(i)) {
                ans.add(i);
            }
        }
        
        return ans;
    }
}
